### ISSUE TITLE
C901: The `max-complexity` linting rule does not appear to work

### ISSUE THREAD

**Issue by grahamlyons** [astral-sh/ruff#11477]
Adding the following into a `ruff.toml` config file and running `ruff check -v` against the file produces no violations, whereas `flake8` with `mccabe` flags the code as too complex.

```toml
[lint.mccabe]
# Flag errors (`C901`) whenever the complexity level exceeds 5.
max-complexity = 3
```

Example code:
```python
def foo(a, b, c):
    if a:
        if b:
            if c:
                return 1
            else:
                return 2
        else:
            return 3
    else:
        return 4
```

Output from `flake8`:
```shell
# flake8 --max-complexity=3 test.py 
test.py:2:1: C901 'foo' is too complex (4)
```

Output from `ruff`:
```shell
# ruff check test.py  -v
[2024-05-20][08:47:12][ruff::resolve][DEBUG] Using configuration file (via parent) at: /workspaces/ruff-test/ruff.toml
[2024-05-20][08:47:12][ruff_workspace::pyproject][DEBUG] `project.requires_python` in `pyproject.toml` will not be used to set `target_version` when using `ruff.toml`.
[2024-05-20][08:47:12][ruff::commands::check][DEBUG] Identified files to lint in: 2.615875ms
[2024-05-20][08:47:12][ruff::commands::check][DEBUG] Checked 1 files in: 31.084µs
All checks passed!
```

Ruff version:
`ruff --version: ruff 0.4.4`

This looks to be a bug but if I'm doing something wrong then please let me know.

**Comment by charliermarsh** [comment]
Did you add `C901` to the list of enabled rules? That rule isn't enabled by default. Something like:

```toml
[lint]
extend-select = ["C901"]
```

**Comment by grahamlyons** [comment]
Thanks for the quick response - that was the problem.

**Comment by charliermarsh** [comment]
No prob.
